src/Display/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import Modal from '.';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { fireEvent, render } from '@testing-library/react';
+
+const props = {
+  title: 'Modal Title',
+  content: 'Modal Content',
+  footer: 'Modal Footer',
+  onClose: jest.fn(),
+};
+
+const OpenedModal = (
+  <Modal
+    title={props.title}
+    isVisible={true}
+    footer={[<button key="button">{props.footer}</button>]}
+    onClose={props.onClose}
+  >
+    <p>{props.content}</p>
+  </Modal>
+);
+
+function setupModal(isVisible: boolean) {
+  const onClose = jest.fn();
+  const ModalComponent = (
+    <Modal
+      title={props.title}
+      isVisible={isVisible}
+      footer={[<button key="button">{props.footer}</button>]}
+      onClose={onClose}
+    >
+      <p>{props.content}</p>
+    </Modal>
+  );
+  const { getByTestId, getByRole } = render(ModalComponent);
+  const modalContainer = getByTestId('modal-container');
+  const modalDialog = getByRole('dialog');
+  return [modalContainer, onClose, modalDialog];
+}
+
+it('<Modal> should render with a title, content, footer and an onClick handler', () => {
+  const ModalSnapshot = renderer.create(OpenedModal).toJSON();
+  expect(ModalSnapshot).toMatchSnapshot();
+});
+
+describe('when modal is opened', () => {
```

Comment:
Hey @franciscrispin , there is another `describe('when modal is opened')` block at `Line 62`. 

I believe this can be merged together.

Reply:
Haha yeah they can. But wanted to separate them for clarity cause they didn't feel like they all fit together. But I guess i'll put them together anyway!